src/main/java/org/jenkinsci/plugins/mesos/MesosCloud.java

```diff
@@ -124,7 +125,29 @@ public MesosCloud(
       boolean checkpoint,
       boolean onDemandRegistration,
       String jenkinsURL) throws NumberFormatException {
-    super("MesosCloud");
+    this("MesosCloud", nativeLibraryPath, master, description, frameworkName, slavesUser, 
+            principal, secret, slaveInfos, checkpoint, onDemandRegistration, jenkinsURL);
+  }
+  
+  /**
+   * Constructor, which also allows to specify a custom name.
+   * @throws NumberFormatException Numeric parameter parsing error
+   * @since TODO
+   */
+  protected MesosCloud(
+      String cloudName,
+      String nativeLibraryPath,
+      String master,
+      String description,
+      String frameworkName,
+      String slavesUser,
+      String principal,
+      String secret,
+      List<MesosSlaveInfo> slaveInfos,
+      boolean checkpoint,
+      boolean onDemandRegistration,
+      String jenkinsURL) throws NumberFormatException {
+    super(cloudName);
```

Comment:
I know you're adding to existing code (and so the pattern has already been set) but it seems like a fluent API on this class would be nicer than these constructors with lots of (what appear to be - no checks on them) optional args.